packages/calcite-components/src/tests/stackedFocusTrap.e2e.ts

```diff
@@ -0,0 +1,127 @@
+import { E2EElement, newE2EPage, E2EPage } from "@stencil/core/testing";
+import { camelCase } from "change-case";
+import { html } from "../../support/formatting";
+import { skipAnimations } from "../tests/utils";
+
+describe("stacked focus-trap components", () => {
+  const componentStack = html`
+    <calcite-sheet id="sheet">
+      <calcite-panel>
+        <calcite-block open></calcite-block>
+        <calcite-button id="sheet-button">Open Modal from Sheet</calcite-button>
+      </calcite-panel>
+    </calcite-sheet>
+
+    <calcite-dialog id="dialog" width-scale="s" scale="s" heading="small scale dialog" description="My description">
+      <p>The small dialog is perfect for short confirmation dialogs or very compact interfaces with few elements.</p>
+      <calcite-button>Back</calcite-button>
+    </calcite-dialog>
+
+    <calcite-modal id="example-modal">
+      <div slot="content">
+        <p>This is an example modal that opens from a Sheet.</p>
+      </div>
+      <calcite-button slot="back">Open Another Modal</calcite-button>
+    </calcite-modal>
+
+    <calcite-modal id="another-modal">
+      <div slot="content">
+        <p>
+          This is an example of a another modal that opens from a modal. This modal an input date picker, a popover and
+          a tooltip.
+        </p>
+        <calcite-popover placement="auto" reference-element="popover-button" closable id="popover">
+          <div>
+            <p>Example Popover.</p>
+            <calcite-label>
+              Input Date Picker
+              <calcite-input-date-picker id="input-date-picker"></calcite-input-date-picker>
+            </calcite-label>
+            <calcite-label>
+              Input Time Picker
+              <calcite-input-time-picker
+                name="calcite-input-time-picker"
+                id="input-time-picker"
+              ></calcite-input-time-picker>
+            </calcite-label>
+          </div>
+        </calcite-popover>
+        <calcite-button id="popover-button">Example Popover</calcite-button>
+        <calcite-button id="tooltip-auto-ref">auto</calcite-button>
+      </div>
+    </calcite-modal>
+  `;
+
+  it("closes a stack of open components sequentially in visual order", async () => {
+    const page = await newE2EPage();
+    await page.setContent(componentStack);
+    await skipAnimations(page);
+
+    async function testStackEscapeSequence(page: E2EPage, pickerType: string) {
+      async function openAndCheckVisibility(element: E2EElement) {
+        element.setProperty("open", true);
+        await page.waitForChanges();
+        expect(await element.isVisible()).toBe(true);
```

Comment:
I think you can omit this since you are testing for openness of all components in `testEscapeAndAssertOpenState`.